### ISSUE TITLE
provider/aws: Support S3 Bucket Cross-Region Replication

### ISSUE THREAD

**Issue by stack72** [hashicorp/terraform#7490]
As requested in #3108

http://docs.aws.amazon.com/AmazonS3/latest/UG/cross-region-replication.html

**Comment by koenijn** [comment]
Indeed we would like to see support for configuring Cross-Region Replication as well.

**Comment by yorrick** [comment]
This would very nice indeed!

**Comment by bpoland** [comment]
+1

**Comment by Ninir** [comment]
Hi guys,

started the work on this! hope it will fit :)

**Comment by ChadScott** [comment]
I don't think this is working properly. I'm following the documentation on the website and I only get `Error putting S3 replication configuration: MalformedXML: The XML you provided was not well-formed or did not validate against our published schema`

If I then `plan` after this failure, it thinks it is enabled but the GUI/API shows it is not.

**Comment by Ninir** [comment]
Hi @ChadScott 
Just took the exact configuration from the [documentation](https://www.terraform.io/docs/providers/aws/r/s3_bucket.html) and it works fine.

Could you please run `TF_LOG=DEBUG terraform apply` and put the output here please?

Thanks!

**Comment by jbz** [comment]
@Ninir I'm having the same issue - I had put replication into place with TF and it worked fine.  I'm now trying to modify the bucket to deal with CloudFront, and *now* it's giving me the same error message @ChadScott got: 

```
Error putting S3 replication configuration: MalformedXML: The XML you provided was not well-formed or did not validate against our published schema
	status code: 400, request id: F2D631F10A79DD1C
```
I've verified that Cross-region replication *is* still enabled on the bucket as it had been prior to the Cloudfront change attempt.  Any advice?

**Comment by Ninir** [comment]
@jbz: it will be fixed in 0.8.3 :) a fix has been provided a few days agi!

**Comment by jbz** [comment]
Ah, many thanks!
